schema/oneof.go

```diff
@@ -44,6 +47,11 @@ func (o OneOfSchema[KeyType]) ApplyScope(scope Scope) {
 	for _, t := range o.TypesValue {
 		t.ApplyScope(scope)
 	}
+	// scope must be applied before we can access the subtypes' properties
+	err := o.validateSubtypeDiscriminatorInlineFields()
+	if err != nil {
+		panic(err)
+	}
```

Comment:
This validation would fail when either the one-of schema or its constituent schema types disagree about the type (string, int) or placement of the discriminator field (i.e. is it inlined with the rest of the subtype's properties).

Reply:
Yeah, it's theoretically an internal problem within a schema, not a direct user issue: but half the world *lives* off the rails. 😆

Reply:
It's not obvious to me whether a `panic()` is the appropriate response here, but I'll assume that it is.